src/components/Payroll/PayrollList/PayrollList.tsx

```diff
@@ -1,11 +1,24 @@
 import { usePayrollsListSuspense } from '@gusto/embedded-api/react-query/payrollsList'
 import { usePaySchedulesGetAllSuspense } from '@gusto/embedded-api/react-query/paySchedulesGetAll'
 import { ProcessingStatuses } from '@gusto/embedded-api/models/operations/getv1companiescompanyidpayrolls'
+import type { Payroll } from '@gusto/embedded-api/models/components/payroll'
 import { PayrollListPresentation } from './PayrollListPresentation'
+import type { PayrollType } from './types'
 import type { BaseComponentInterface } from '@/components/Base'
 import { BaseComponent } from '@/components/Base'
 import { componentEvents } from '@/shared/constants'
 
+const selectPayrollType = ({
```

Comment:
There was an interesting code design decision here where we need to interpret payroll type from the payroll data because the type doesn't exist on the API payload.

What I chose to do was to have the Presentation layer expect the problem to be solved before it received it as Props. PayrollList then takes a payroll and creates a projection from the payroll with synthesized props from the API payload. The synthesized prop has to be extracted to `types` since both PayrollList and PayrollLisPresentation both need to depend on it.